src/java.desktop/macosx/native/libawt_lwawt/awt/AWTEvent.m

```diff
@@ -291,7 +291,7 @@
         //kCGSFlagsMaskAppleLeftAlternateKey,
         //kCGSFlagsMaskAppleRightAlternateKey,
         58,
-        0,
+        61,
```

Comment:
Looks like the only change required to make right option key working is adding in the rightKeyCode (61) here. I don't think it is required to revert the `leftAltKeyPressed` changes.

Reply:
This seems correct. I updated it to only include the key code. Re-ran my test and TestAltUpDown, and both passed still.